src/Aspire.Dashboard/Components/Pages/Resources.razor

```diff
@@ -10,25 +10,75 @@
+    </PageTitleSection>
+
+    <ToolbarSection>
+        @if (ViewportInformation.IsDesktop)
+        {
+            <FluentButton id="typeFilterButton" slot="end"
+                          Appearance="@(AreAllTypesVisible is true ? Appearance.Stealth : Appearance.Accent)"
+                          IconEnd="@(new Icons.Regular.Size24.Filter())"
+                          @onclick="() => _isTypeFilterVisible = !_isTypeFilterVisible"
+                          Title="@(AreAllTypesVisible is true ? Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFilterAllVisible)] : Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFiltered)])"
+                          aria-label="@(AreAllTypesVisible is true ? Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFilterAllVisible)] : Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFiltered)])"/>
+        }
+        else
+        {
+            <div>
+                <h4>@(AreAllTypesVisible is true ? Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFilterAllVisible)] : Loc[nameof(Dashboard.Resources.Resources.ResourcesTypeFiltered)])</h4>
+
+                <FluentStack Orientation="Orientation.Vertical">
+                    <FluentCheckbox
+                        Label="@ControlsStringsLoc[nameof(ControlsStrings.All)]"
+                        ThreeState="true"
+                        ShowIndeterminate="false"
+                        ThreeStateOrderUncheckToIntermediate="true"
+                        @bind-CheckState="AreAllTypesVisible"/>
+                    @foreach (var (resourceType, _) in _allResourceTypes)
+                    {
+                        var isChecked = _visibleResourceTypes.ContainsKey(resourceType);
+                        <FluentCheckbox
+                            Label="@resourceType"
+                            @bind-Value:get="isChecked"
+                            @bind-Value:set="c => OnResourceTypeVisibilityChangedAsync(resourceType, c)"/>
+                    }
+                </FluentStack>
+            </div>
+        }
+
+
+            @if (ViewportInformation.IsDesktop)
+            {
+                <FluentSearch Placeholder="@ControlsStringsLoc[nameof(ControlsStrings.FilterPlaceholder)]"
+                              Immediate="true"
+                              @bind-Value="_filter"
+                              slot="end"
+                              @bind-Value:after="HandleSearchFilterChangedAsync"/>
+            }
+            else
+            {
+                <div>
+                    <FluentSearch Placeholder="@ControlsStringsLoc[nameof(ControlsStrings.FilterPlaceholder)]"
+                                  Immediate="true"
+                                  @bind-Value="_filter"
+                                  slot="end"
+                                  Label="@ControlsStringsLoc[nameof(ControlsStrings.FilterPlaceholder)]"
+                                  @bind-Value:after="HandleSearchFilterChangedAsync"/>
```

Comment:
Can this duplication be avoided?

Reply:
The fact that one control was changed was intentional, but I can do that inline

Reply:
I know the difference here is intentional. I mean future changes.